Store: Stats: Add Referrer Page

psealock opened the pull request:
## Add All Referrers Page 
This PR is an incremental addition to the All Referrers Page, which is behind a feature flag. None of this code will be accessible outside of development. This code adds the following:
* Route (behind a feature flag)
* Period Navigation
* Injected data to the page
* Wrap data in `<Module>` element
* Print data in a table (temporarily)

<img width="840" alt="screen shot 2018-03-19 at 1 48 57 pm" src="https://user-images.githubusercontent.com/1922453/37573396-50191e46-2b7c-11e8-8d1a-d4f015b4470a.png">

### Testing
* Test the link to Store Referrers page from Store Stats
* Make sure the navigation works (units, arrows to switch periods)
* See data loading and displayed in tabular form

### Following Pull Requests
* Create a search box to select a specific referrer
* Add a main chart

justinshreve commented on client/extensions/woocommerce/app/store-stats/referrers/index.js:
```diff
@@ -0,0 +1,99 @@
+} ) => {
+	const unitSelectedDate = getUnitPeriod( selectedDate, unit );
+	const selectedData = find( data, d => d.date === unitSelectedDate ) || { data: [] };
+	const sortedData = sortBy( selectedData.data, d => -d.sales );
```
We do this in a few places now (grab a row by date, and then order by sales): Here, your stats widget, and my dashboard widget. Maybe we should have a helper function for referrer data? It could optionally take a `limit` param since we slice the results sometimes too.

justinshreve commented on client/extensions/woocommerce/app/store-stats/referrers/index.js:
```diff
@@ -0,0 +1,99 @@
+				statsType={ STAT_TYPE }
+				path={ path }
+				querystring={ querystring }
+				title="Store Referrers"
```
Needs a `translate`

justinshreve commented on client/extensions/woocommerce/app/store-stats/store-stats-period-nav/index.js:
```diff
@@ -0,0 +1,90 @@
+import Intervals from 'blocks/stats-navigation/intervals';
+import DatePicker from 'my-sites/stats/stats-date-picker';
+
+const getGoBackFn = ( path, querystring ) => {
```
Would `page.back` be useful here? (I haven't used it myself) or maybe [window.history.back](https://github.com/Automattic/wp-calypso/blob/112c7f3c62e8c089dd2756fe40de99fb90e50b4e/client/reader/header-back/index.jsx#L6)?

psealock replied:
I replaced this logic with `window.history.back`. The only place this might fail is if someone links to this page from an external source or pastes the link directly. The experience will be a bit odd, but perhaps not unexpected. Otherwise, it works well and reduces some logic and code

@greenafrican, what do you think? Modification in 827775088a6d1dbefc3cc7945dec77c4144d6395, I can easily revert

justinshreve commented on client/extensions/woocommerce/app/store-stats/controller.js:
```diff
@@ -75,24 +83,50 @@ export default function StatsController( context, next ) {
+			asyncComponent = (
+				<AsyncLoad
+					/* eslint-disable wpcalypso/jsx-classname-namespace */
+					placeholder={ <StatsPagePlaceholder className="woocommerce" /> }
```
Minor - maybe define a placeholder variable so you only have to do the eslint rules once. Looks like it's the same placeholder in all three cases.

psealock commented:
Thanks for the review @justinshreve 

### Fixes
* Fixed type `statsType` → `statType`
* Added `translate`
* Cleaned up `placeholder` prop declaration in `controller.js`
* Pulled the sorting and trimming to a helper function and included relevant tests
